### ISSUE TITLE
Footlight Minimal Search widget

### ISSUE THREAD

**Issue by sahalali** [culturecreates/minimal-footlight-search-widget#10]

**Comment by sahalali** [comment]
@sahalali Please find the attachment containing the icons for the calendar.


[Calendrier-20230719T173048Z-001.zip](https://github.com/culturecreates/minimal-footlight-search-widget/files/12102419/Calendrier-20230719T173048Z-001.zip)

**Comment by troughc** [comment]
comments from DEMO
Issue
- Single dates: sometimes there is a space between on left
- Single dates: when you select single date - the wrong events (wrong dates) are displayed - not seeing the right dates
- When you click effacer -  the calendar on left should reset to the current date
- When you click ‘load all results button at bottom you should display all the events (limit may not being passed in the payload - comment from Abhishek)
- Looks like the label fallback configuration is missing (comment from Gregory)


Mobile
- Date range selection - looks good
- Single date shows too many dates - not passing the end date (Siam’s comment)
- In mobile Search bar: can only show date or search term - search takes precedent.

Organizations
- Date should not be shown
- Date as a parameter should not be sent with organizations
- 

Questions for MINIMAL
Image
- When image missing, do you want to use/provide a default image? 
- How to display the date and search term in mobile? Can we put it on 2 lines

**Comment by sahalali** [comment]
![Image](https://github.com/culturecreates/minimal-footlight-search-widget/assets/139197166/5d916065-95eb-4763-87d4-2ea1a565c930)
![Image](https://github.com/culturecreates/minimal-footlight-search-widget/assets/139197166/a6258a07-a145-4a04-8e4c-dd6b9a3de2fe)

**Comment by troughc** [comment]
Caitlin's early feedback from trying the new search widget in Gregory's staging site. 

1. When I click an event in the list, no event is loaded (this is the same issue as when you click 'voir tous les résultats' as discussed during the demo).
2. Are you supposed to be able to scroll the event list display on the left? I cannot. When I load a date range,  I can see 4 events displayed and the voir tous les resultats with a number (see screenshot). I cannot scroll the list on the left to see more events. 


![Image](https://github.com/culturecreates/minimal-footlight-search-widget/assets/64489312/778d93f2-f0eb-456e-91ec-edcf1bd14dac)

3. already noted is that the single date option (to select a single date isn't working as expected. you need an end date I think?
4. the Clear option (EFFACER) seems to be working okay but I noticed one thing:  When i click effacer, the number of events displayed next to 'voir tous les résultats' is 181 and NO DATE is highlighted in the calendar (on the right). How do I know what this number (181) is referring to? In contrast, if I enable the single date option and select a date in the future (say in August) the number of events  next to 'voir tous les résultats' is different (depending on the date). So my question is: do we need to show (on the calendar) what date or date range is being used to generate the number of events when nothing is selected? (I hope this is clear)?
5. How do you clear out a search term? Do you have to select it and then delete it?
6. for Organizations section: you need to remove the word "prochainement" from the top of the list, as well as the strange spacing when there are fewer than 4 results (as discussed in the meeting). To see this behavior of fewer results, you can search on "ACAD"

**Comment by troughc** [comment]
@SyamBabu-M here is my guidance in writing:
- please have the search for a specific date (single date) display results for that specific date (for example, the visitor clicks August 10 and gets results for August 10); the user searches for 'Danse' on August 10 and gets results that include the term 'Danse' that are happening on Aug. 10
NOTE: when no results are found, the date text that is displayed will be correct (because the single date is displayed under 'aucune evenement displonible')
- Date range seems to be working well
- By default, the way you highlight the current date (today's date) should be different from any dates you select (either a single date or date range). Right now you have the underline to diffentiate it, which is good. It would only be yellow if the user/visitor selected it. 
Hope this helps.

**Comment by troughc** [comment]
When no events are available the text has a typo with : Aucun Événements disponible
It should be:

Aucun événement disponible

**Comment by SyamBabu-M** [comment]
@troughc Hi, thanks for the detailed summary in writing, all the points mentioned above are taken care of.
1: Fixed the spelling error in the message when no events are available.
2: Changed the default highlight color on today's date. 
3: Changed the query params to include the end date so that the result will be for that specific date only

**Comment by troughc** [comment]
@SyamBabu-M  I am looking at it in Gregory's test area here http://demo.staging-tout-culture.s3-website.ca-central-1.amazonaws.com/ '
Some things don't seem to have changed? But maybe it's because I am looking in the wrong place. Here is my test results:

**Point 2** - looks good 

**point 1**, I am still seeing the spelling error:
To reproduce:
Select a date far into the future (2025 Aug 10 for example) when there are no dates available.
- On the left side the text displays Aucun Événements disponible
- the text should be:  Aucun événement disponible
(Please note the change is to the second word -  capitalization and no 's' at the end)

**Point 3: i**t doesn't look like it is displaying only the events that are happening on the one selected date. 
Steps to reproduce:
- Select a date in the calendar. 
- The list updates to include all future events (from the selected date going forward).

**Comment by troughc** [comment]
Thanks for merging the update @SyamBabu-M 
New TEST on Chrome
Point 1: PASSED
Point 3: Looking better but still has some issues: there are some wrong **events that are event series** that are being included in single date searches. These wrong events don't have any sub event on my selected date. 
to reproduce:
- select  August 23, 2023 in the calendar view
- The results update (18 events). This is good because i can see that the results are closer to what I expect. However, there are events included that are not happening on august 23.
- Open these events to see that they don't have a sub event on August 23: Jam Trad au Quai des Légendes (there is an event on August 24).
- Select August 24, 2023 in the calendar
- open this event: Le bureau de Charles Symmes - Chambre d'évasion  This one has no event on August 24 (it has one on august 23 and august 25)

Hope this helps with troubleshooting.

**Comment by sahalali** [comment]
@troughc Point 3 is a known issue in the backend. 
Currently, the subEvent dates are not considered while filtering but the startDate and endDate of the parent Event.

@saumier I am not sure we have an open issue for this. If not, can we open an issue?

**Comment by saumier** [comment]
@sahalali Yes. This is issue https://github.com/culturecreates/footlight-calendar-api/issues/537

**Comment by troughc** [comment]
@SyamBabu-M for largest screens (highest res) fix display of organizations so the names are left aligned as per the trello feedback.

**Comment by troughc** [comment]
@SyamBabu-M request to change the wording of message we display when no events are available. This message:


![Image](https://github.com/culturecreates/minimal-footlight-search-widget/assets/64489312/36c8d86b-8d4f-4b96-9ca1-6b97fb80ed2b)




Please change to:

French:  Aucun événement présentement
English:  No events at the moment

**Comment by troughc** [comment]
message for workshops in French: Aucun ateliers présentement

**Comment by troughc** [comment]
Gentle reminder @SyamBabu-M 

Please fix the messages that appear when no result is returned. 
Pay careful attention to capitalization of each word in the sentence. Only the first word is capitalized. 

You need to make a changes to both French and English messages.

**FRENCH fixes:**

For workshops (ateliers) - note the capitalization change as well
Replace 'Aucun Ateliers disponible' with the following:

Aucun ateliers présentement 

For Events (événements) - please pay attention to accents on letters) 
Replace 'Aucun evénement disponible' with the following:

Aucun événement présentement 

For organizations (organisations) - note the capitalization change
Replace 'Aucun Organisations disponible' with the following

Aucun organisations disponible




**ENGLISH fixes (only need to fix capitalization of second word):**

For workshops: Replace 'No Workshops at the moment' with the following:

No workshops at the moment

For organizations: Replace 'No Organizations are available' with the following

No organizations are available


cc @sahalali

**Comment by SyamBabu-M** [comment]
@troughc Noted.

**Comment by troughc** [comment]
@SyamBabu-M a little research to prevent the zoom in on input elements (when you focus on a field). cc @saumier

**Comment by troughc** [comment]
Tested in Gregory's test site on CHROME
everything I tested is working great: PASS

That said, I stumbled on one question I have for you @saumier regarding the following: 
eventId=648cd2ed5526a700645056d7


The event types include both "Festival" and "Workshop | Art and Craft". 

The event only appears in Workshops (atelier) search results when I search on Festival (It is not included in events searches). 

Is that the expected behavior? Logically, it could be, one could argue that festival is a secondary category to a type of workshop. However, Festival has it's own separate entry in the controlled vocabulary for event type (it is not a sub-category to 'course' or 'workshop').
 @saumier I would like your opinion on this.
